server/src/main/java/io/crate/execution/MultiPhaseExecutor.java

```diff
@@ -45,14 +50,21 @@ public static CompletableFuture<SubQueryResults> execute(Map<LogicalPlan, Select
                                                              DependencyCarrier executor,
                                                              PlannerContext plannerContext,
                                                              Row params) {
+        var ramAccounting = ConcurrentRamAccounting.forCircuitBreaker(
+            "multi-phase",
+            executor.circuitBreaker(HierarchyCircuitBreakerService.QUERY),
+            plannerContext.transactionContext().sessionSettings().memoryLimitInBytes()
+        );
+
```

Comment:
Might be worth checking performance impact of this - and if it's worth wrapping into `BlockBasedRamAccounting` (but _per_ consumer, further below) to reduce the amount of synchronization.